fixed minor typo: `it this` -> `this`

arteniioleg opened the pull request.

cubbimew commented:
just noticed, this went into the gh-pages branch. Could you re-apply to master branch?

arteniioleg commented:
I can't.

```bash
$ git cherry-pick e854e47
error: could not apply e854e47... fixed minor typo: `it this` -> `this` (#1077)
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'
```

The conflicts

```
diff --cc CppCoreGuidelines.md
index d81e8fc,7a7346e..0000000
--- a/CppCoreGuidelines.md
+++ b/CppCoreGuidelines.md
@@@ -12414,43 -12373,79 +12414,55 @@@ whereas `if (p != nullptr)` would be a 
  
  ##### Example
  
++<<<<<<< HEAD
 +This rule is especially useful when a declaration is used as a condition
++=======
+     void f(int* p)
+     {
+         int x = *p; // BAD: how do we know that p is valid?
+     }
+ 
+ There is a huge amount of such code.
+ Most works -- after lots of testing -- but in isolation it is impossible to tell whether `p` could be the `nullptr`.
+ Consequently, this is also a major source of errors.
+ There are many approaches to dealing with this potential problem:
++>>>>>>> e854e47... fixed minor typo: `it this` -> `this` (#1077)
  
 -    void f1(int* p) // deal with nullptr
 -    {
 -        if (p == nullptr) {
 -            // deal with nullptr (allocate, return, throw, make p point to something, whatever
 -        }
 -        int x = *p;
 -    }
 +    if (auto pc = dynamic_cast<Circle>(ps)) { ... } // execute is ps points to a kind of Circle, good
  
 -There are two potential problems with testing for `nullptr`:
 +    if (auto pc = dynamic_cast<Circle>(ps); pc != nullptr) { ... } // not recommended
  
 -* it is not always obvious what to do what to do if we find `nullptr`
 -* the test can be redundant and/or relatively expensive
 -* it is not obvious if the test is to protect against a violation or part of the required logic.
 +##### Example
  
 +Note that implicit conversions to bool are applied in conditions.
 +For example:
  
 -    void f2(int* p) // state that p is not supposed to be nullptr
 -    {
 -        assert(p != nullptr);
 -        int x = *p;
 -    }
 +    for (string s; cin >> s; ) v.push_back(s);
  
 -This would carry a cost only when the assertion checking was enabled and would give a compiler/analyzer useful information.
 -This would work even better if/when C++ gets direct support for contracts:
 +This invokes `istream`'s `operator bool()`.
  
 -    void f3(int* p) // state that p is not supposed to be nullptr
 -        [[expects: p != nullptr]]
 -    {
 -        int x = *p;
 -    }
 +##### Example, bad
  
 -Alternatively, we could use `gsl::not_null` to ensure that `p` is not the `nullptr`.
 +It has been noted that
  
 -    void f(not_null<int*> p)
 -    {
 -        int x = *p;
 -    }
 +    if(strcmp(p1, p2)) { ... }   // are the two C-style strings equal? (mistake!)
  
 -These remedies take care of `nullptr` only.
 -Remember that there are other ways of getting an invalid pointer.
 +is a common beginners error.
 +If you use C-style strings, you must know the `<cstring>` functions well.
 +Being verbose and writing 
  
 -##### Example
 +    if(strcmp(p1, p2) != 0) { ... }   // are the two C-style strings equal? (mistake!)
  
 -    void f(int* p)  // old code, doesn't use owner
 -    {
 -        delete p;
 -    }
 +would not save you.
  
 -    void g()        // old code: uses naked new
 -    {
 -        auto q = new int{7};
 -        f(q);
 -        int x = *q; // BAD: dereferences invalid pointer
 -    }
 +##### Note
  
 -##### Example
 +The opposite condition is most easily expressed using a negation:
  
 -    void f()
 -    {
 -        vector<int> v(10);
 -        int* p = &v[5];
 -        v.push_back(99); // could reallocate v's elements
 -        int x = *p; // BAD: dereferences potentially invalid pointer
 -    }
 +    // These all mean "if `p` is `nullptr`"
 +    if (!p) { ... }           // good
 +    if (p == 0) { ... }       // redundant `!= 0`; bad: don't use `0` for pointers
 +    if (p == nullptr) { ... } // redundant `== nullptr`, not recommended
  
  ##### Enforcement
  
```

I can make it as a separate/manual change and pull request for `master`, but you will get conflict when will merge `master` in `gh-pages`. To do so?

johelegp commented:
If the workflow is still the same, the gh-pages branch is manually generated from the master branch from time to time. Work on the master branch will eventually appear in the gh-pages. If you can't cherry-pick your change, you'll have to do it manually as you mention.
